obspy/signal/polarization.py

```diff
@@ -86,48 +87,482 @@ def eigval(datax, datay, dataz, fk, normf=1):
+    :type spoint: :class:`numpy.ndarray`
+    :param stime: start time of the analysis
+    :type stime: :class:`~obspy.core.utcdatetime.UTCDateTime`
+    :param etime: end time for the analysis
+    :type etime: :class:`~obspy.core.utcdatetime.UTCDateTime`
+    :returns: azimuth, incidence, rectilinearity planarity, and ellipticity
+    """
+    W = 3.0
+    stream.sort(reverse=True)
+    Z = stream[0].data.copy()
+    N = stream[1].data.copy()
+    E = stream[2].data.copy()
+
+    Zi = instantaneous_frequency(Z, fs)
+    Za = signal.hilbert(Z)
+    Ni = instantaneous_frequency(N, fs)
+    Na = signal.hilbert(N)
+    Ei = instantaneous_frequency(E, fs)
+    Ea = signal.hilbert(E)
+    res = []
+
+    offset = int(3 * fs / flow)
+    while True:
+        adapt = int(3. * W * fs / (Zi[offset] + Ni[offset] + Ei[offset]))
+        # in order to account for errors in the inst freq estimation
+        if adapt > int(3 * fs / flow):
+            adapt = int(3 * fs / flow)
+        if adapt < int(3 * fs / fhigh):
+            adapt = int(3 * fs / fhigh)
+        adapt /= 2
+        adapt = (2 * adapt) + 1
+        newstart = stime + offset / fs
+        if (newstart + (adapt / 2) / fs) > etime:
+            break
+
+        Zx = Za[int(spoint[2] + offset - adapt / 2):
+                int(spoint[2] + offset + adapt / 2)]
+        Nx = Na[int(spoint[1] + offset - adapt / 2):
+                int(spoint[1] + offset + adapt / 2)]
+        Ex = Ea[int(spoint[0] + offset - adapt / 2):
+                int(spoint[0] + offset + adapt / 2)]
+        Zx -= Zx.mean()
+        Nx -= Nx.mean()
+        Ex -= Ex.mean()
+
+        covmat = np.zeros([3, 3], dtype=np.complex128)
+
+        covmat[0][0] = np.dot(Ex, Ex.conjugate())
+        covmat[0][1] = np.dot(Ex, Nx.conjugate())
+        covmat[1][0] = covmat[0][1].conjugate()
+        covmat[0][2] = np.dot(Ex, Zx.conjugate())
+        covmat[2][0] = covmat[0][2].conjugate()
+        covmat[1][1] = np.dot(Nx, Nx.conjugate())
+        covmat[1][2] = np.dot(Zx, Nx.conjugate())
+        covmat[2][1] = covmat[1][2].conjugate()
+        covmat[2][2] = np.dot(Zx, Zx.conjugate())
+
+        eigvec, eigenval, v = (np.linalg.svd(covmat))
+
+        def fun(x):
+            return 1. - math.sqrt(
+                ((eigvec[0][0] * (math.cos(x) + math.sin(x) * 1j)).real) ** 2 +
+                ((eigvec[1][0] * (math.cos(x) + math.sin(x) * 1j)).real) ** 2 +
+                ((eigvec[2][0] * (math.cos(x) + math.sin(x) * 1j)).real) ** 2)
+
+        final = fminbound(fun, 0.0, math.pi, full_output=True)
+        X = 1. - final[1]
+        ellip = math.sqrt(1.0 - X ** 2) / X
+        # rectilinearity defined after Montalbetti & Kanasewich, 1970
+        rect = 1. - np.sqrt(eigenval[1] / eigenval[0])
+        # planarity defined after Jurkevics, 1988
+        plan = 1. - (2.0 * eigenval[2] / (eigenval[1] + eigenval[0]))
+
+        azimuth = 180 * math.atan2(eigvec[0][0].real, eigvec[1][0].real) / \
```

Comment:
The only difference with `flinn` is this one doesn't use the `degrees` function.